addons/crm/models/res_partner.py

```diff
@@ -63,3 +63,15 @@ def action_view_opportunity(self):
         else:
             action['domain'] = [('partner_id.id', '=', self.id)]
         return action
+
+    def _fields_sync(self, values):
+        # sync salesteam with parent
+        if not self.team_id and not values.get('team_id') and values.get('parent_id') and self.company_type == 'person':
```

Comment:
As it's done just after the write / create, I think you can just do
```suggestion
        if not self.team_id and self.parent_id and self.company_type == 'person':
```